libs/trustchain/src/HWDeviceProvider.ts

```diff
@@ -1,17 +1,33 @@
+import { from, lastValueFrom } from "rxjs";
 import { UserRefusedOnDevice } from "@ledgerhq/errors";
 import { ApduDevice } from "@ledgerhq/hw-trustchain/ApduDevice";
-import Transport, { StatusCodes, TransportStatusError } from "@ledgerhq/hw-transport";
+import { StatusCodes, TransportStatusError } from "@ledgerhq/hw-transport";
 import { crypto, device } from "@ledgerhq/hw-trustchain";
 import api from "./api";
 import { genericWithJWT } from "./auth";
-import { AuthCachePolicy, JWT, TrustchainDeviceCallbacks } from "./types";
+import { AuthCachePolicy, JWT, TrustchainDeviceCallbacks, WithDevice } from "./types";
 
 export class HWDeviceProvider {
+  /**
+   * TODO withDevice should be imported statically from @ledgerhq/live-common/hw/deviceAccess
+   *
+   * but ATM making @ledgerhq/live-common a dependency of @ledgerhq/trustchain causes:
+   * > Turbo error: Invalid package dependency graph: cyclic dependency detected:
+   * >   @ledgerhq/trustchain,@ledgerhq/live-wallet,@ledgerhq/live-common
+   *
+   * Maybe hw/deviceAccess.ts and hw/index.ts could be moved to @ledgerhq/devices
+   * This would break the cyclic dependency as @ledgerhq/live-common would depend on @ledgerhq/devices
+   * but not the other way around.
```

Comment:
I prefer to keep it as is, otherwise it involves much work to be done. ( + I prefer that you inject the withDevice as a parameter rather than importing it in the trustchain sdk)

Reply:
the idea to move hw/* and notably the deviceAccess / index files of it to a library is a good idea, but i think it's the scope of device team tho. also the impact is quite important in that many parts depends on this. but this could be discussed with that team.